OpenSim/Common/DataTable.h

```diff
@@ -1350,26 +1355,42 @@ class DataTable_ : public AbstractDataTable {
 
     // Split element into constituent components and append the components to
     // the given vector. For example Vec3 has 3 components.
-    template<int N>
+    template<int N, typename Iter>
     static
-    void splitElementAndPushBack(std::vector<double>& row,
-                                 const SimTK::Vec<N>& elem) {
-        for(unsigned i = 0; i < N; ++i)
-            row.push_back(elem[i]);
+    void splitElementAndAppend(Iter begin, Iter end, 
+                               const SimTK::Vec<N>& elem) {
+        for(unsigned i = 0; i < N; ++i) {
+            OPENSIM_THROW_IF(begin == end,
+                InvalidArgument,
+                "Iterators do not produce enough elements."
+                "Expected: " + std::to_string(N) + " Received: " +
+                std::to_string(i));
+
+            *begin++ = elem[i];
+        }
     }
     // Split element into constituent components and append the components to 
-    // the given vector. . For example Vec<2, Vec3> has 6 components.
-    template<int M, int N>
+    // the given vector. For example Vec<2, Vec3> has 6 components.
+    template<int M, int N, typename Iter>
     static
-    void splitElementAndPushBack(std::vector<double>& row,
-                                 const SimTK::Vec<M, SimTK::Vec<N>>& elem) {
-        for(unsigned i = 0; i < M; ++i)
-            for(unsigned j = 0; j < N; ++j)
-                row.push_back(elem[i][j]);
+    void splitElementAndAppend(Iter begin, Iter end, 
+                               const SimTK::Vec<M, SimTK::Vec<N>>& elem) {
+        for(unsigned i = 0; i < M; ++i) {
+            for(unsigned j = 0; j < N; ++j) {
+                OPENSIM_THROW_IF(begin == end,
+                    InvalidArgument,
+                    "Iterators do not produce enough elements."
+                    "Expected: " + std::to_string(M * N) +
```

Comment:
Add space; `" Expected: "`.

Reply:
Actually, I don't think I need this space if I add after the period above.